compiler/driver.ml

```diff
@@ -541,14 +541,27 @@ module Commands = struct
     (* Strictly type-checking could stop here, but we also want this pass to
        check full name-resolution and cycle detection. These are checked during
        translation to dcalc so we run it here and drop the result. *)
-    let _prg = Dcalc.From_scopelang.translate_program prg in
+    let prg = Dcalc.From_scopelang.translate_program prg in
+    let prg =
```

Comment:
This re-typing should only be done if required by `check_invariants`: in normal use everything that needs to be checked at this point has already been done.

Reply:
done!